setup.py

```diff
@@ -197,13 +197,20 @@ def run(self):
     'testpath', 
 ]
 
-extras_require = setuptools_args['extras_require'] = {
+extra_requirements = {
     # FIXME: tests still require nose for some utility calls,
     # but we are running with pytest
-    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client'],
+    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client>=4.2'],
     'serve': ['tornado>=4.0'],
     'execute': ['jupyter_client>=4.2'],
 }
+s = [] 
+[s.extend(values) for values in extra_requirements.values()]
```

Comment:
I don't like using list comprehensions for side effects. I'd either do this with a regular for loop:

```python
for reqs in extra_requirements.values():
    s.extend(reqs)
```

Or get clever with `sum`:

```python
s = sum(extra_requirements.values(), [])
```